Create new GMSH module

aidanjungo opened the pull request:
For more details #144

GovoniT committed:
Create file structure (#160)

GovoniT committed:
Gmsh module basic function (#163)

* working exportbrep.py

* add test for exportbrep.py

* add generatemesh.py, add gmsh to the environment

* add a test for generatemesh.py and a gitignore for the tests folder

* complete description of the script

aidanjungo committed:
Few modif to make the code more "CEASIOMpy" compatible

Commit:
add option for gmsh gui display and aircraft principle parts  mesh size

Commit:
add a more robust remapping of the surface and points +  add color code for aircraft part in the mesh

Commit:
small correction on some log.info format, + modification of the test for generatemesh.py

aidanjungo commented on ceasiompy/utils/xpath.py:
```diff
@@ -73,3 +73,6 @@
 PYTORNADO_XPATH = "/cpacs/toolspecific/pytornado"
+
+# GMSH
+GMSHFILE_XPATH = "/cpacs/toolspecific/gmsh"
```
It should be:

```suggestion
GMSHFILE_XPATH = "/cpacs/toolspecific/CEASIOMpy/mesh/gmsh"
```

aidanjungo commented on ceasiompy/CPACS2GMSH/func/generategmesh.py:
```diff
@@ -0,0 +1,534 @@
+        if "wing" in part.name:
+            return 1
+        if "engine" in part.name:
+            return 3
```
It could be a property of the AircraftPart Class no?

And it will probably necessary to have more flexibility for the names, e.g. ["Fuselage", "fuslage", "Fus", "fus", ..]

GovoniT replied:
These names are defined (by us) when we export the .brep file with tigl, so they are always the same and it is intended that we can identify each part of the plane clearly: fuselage1, fuselage2, wing1, wing1_m, ...

For the scores, of course, this could be a class property, I will move this piece of code where the aircraft parts are defined.

aidanjungo commented on ceasiompy/CPACS2GMSH/func/generategmesh.py:
```diff
@@ -0,0 +1,534 @@
+    part1_score = give_score(part1)
+    part2_score = give_score(part2)
+
+    if part1_score >= part2_score:
```
With the previous commant it could be:

```suggestion
    if part1.score >= part2.score:
```

aidanjungo commented on ceasiompy/CPACS2GMSH/func/generategmesh.py:
```diff
@@ -0,0 +1,534 @@
+                matrix[i][j] = aircraft_parts[i].surfaces.intersection(aircraft_parts[j].surfaces)
+            if matrix[i][j]:  # check if not empty
+                # One of the two part need to transfert the duplicated surface to the other
+                Flag_duplicated_surface = True
```
Probably possible to to something simpler and faster with numpy

Commit:
resolve E501 line too long

Commit:
add spacing in the code

Commit:
mesh file accesible now for the forklow

Commit:
Add symmetry option (beta) and farfield size factor option

Commit:
Correct pyflake erro

Commit:
Correct pyflake error

aidanjungo commented:
With the default D150, on Mac I got this error:

```bash
INFO     - ----- Start of cpacs2gmsh.py -----
TIXI handle has been created for /CEASIOMpy/ceasiompy/../WKDIR/Test01/Workflow_036/01_SettingsGUI/ToolOutput.xml.
TIXI handle has been created for /CEASIOMpy/ceasiompy/../WKDIR/Test01/Workflow_036/01_SettingsGUI/ToolOutput.xml.
Exporting brep file...
Exporting brep file...
Exporting brep file...
Exporting brep file...
Exporting brep file...
Exporting brep file...
INFO     - Finding boundary boxes :wing2_m
[1]    9132 segmentation fault  python run_ceasiompy.py -cfg ../WKDIR/Test01/ceasiompy.cfg
```

I tried to change a bit the mesh size but the error stays the same, @GovoniT do you have an idea where it could come from?

GovoniT commented:
Realy weird because "Finding boundary boxes" is a very basic operation,
I think I have to install a Mac virtual machine and try it by myself to find the issue

Commit:
small correction

Commit:
pyflake correction

Commit:
pyflake correction

Commit:
Add wing classification and structured meshing section (Beta)

Commit:
pyflake correction

Commit:
try to solve issue with github pytest in pull request

Commit:
Classification of the wing part untruncated/truncated profiles working

Commit:
Rework of wingclassification in another script file

Commit:
delete log file

Commit:
upgrade wing classification

Commit:
add surface of wing section classification

Commit:
add robustness to wingclassification and add advancemeshing.py (beta)

Commit:
Complet rework of the surface assignation algorithm, gain of stability and accuracy in the surfaces assignation, now direct child to parent relation is possible between each entites in the model
Rework of the symmetry operation, gain in stability, it is now possible to chose the normal vector of the plane of symmetry

Commit:
add robustness to wingclassification, add general background mesh function

Commit:
add function description in generategmesh new functions

Commit:
add test for the assignation of the parts and their entitites, add test for the supression of symetric part when symmetry op. is applied

Commit:
add pylon integration

Commit:
add skin mesh near the aircraft

Commit:
start engine integration

Commit:
engine integration working in generatemesh.py but engines have to be converted mannualy using engineconversion.py

Commit:
add engine closing with boundary inlet and outlet detection (beta) add rotatingdisk generation (beta)

Commit:
add testing of the wingclassification

Commit:
add box in advance meshing, add advance meshing parameters in the GUI settings

Commit:
small fix on the final domain volume assignation

aidanjungo commented on ceasiompy/CPACS2GMSH/__specs__.py:
```diff
@@ -0,0 +1,122 @@
+    gui=True,
+    gui_name="Advanced mesh generation",
+    gui_group="Advanced mesh parameters",
+)
```
This option can be removed and we can use 1.0 by if we don't want any refinement

aidanjungo commented on ceasiompy/CPACS2GMSH/cpacs2gmsh.py:
```diff
@@ -0,0 +1,141 @@
+TODO:
+
+    * Things to improve ...
+    * Things to add ...
```
You can update these info

Commit:
typos

Commit:
typos + remove not finished script for a first branch merging

Commit:
small correction on test_wingclassification.py

Commit:
rebase d150

 Author:    Tony Govoni <tony.govoni@epfl.ch>

 interactive rebase in progress; onto 264e443
 Last commands done (40 commands done):
    pick 532590f small correction on test_wingclassification.py
    pick 036967a rebase d150
 No commands remaining.
 You are currently rebasing branch 'gmsh_module' on '264e443'.

 Changes to be committed:
	modified:   test_files/CPACSfiles/D150_simple.xml

 Untracked files:
	ceasiompy/SU2Run/files/config_template_v3.0.0.cfg
	ceasiompy/SU2Run/files/config_template_v7.0.0.cfg

aidanjungo committed:
fix D150

aidanjungo committed:
Add template for GMSH docs

aidanjungo commented on ceasiompy/CPACS2GMSH/func/advancemeshing.py:
```diff
@@ -0,0 +1,288 @@
+
+    sigmoid = False
+    include_boundary = True
+    surfaces_wing = wing_part.surfaces_tags
```
If use only at one place, the variable could be directly in the code

aidanjungo commented on ceasiompy/CPACS2GMSH/func/exportbrep.py:
```diff
@@ -0,0 +1,170 @@
+Python version: >=3.7
+
+| Author: Name
+| Creation: YEAR-MONTH-DAY
```
to fill

Commit:
typos

aidanjungo commented on ceasiompy/CPACS2GMSH/func/exportbrep.py:
```diff
@@ -0,0 +1,170 @@
+                export_shapes([pylon_m], str(brep_file))
+                need_symmetric_engine = True
+            else:
+                need_symmetric_engine = False
```
```suggestion
            need_symmetric_engine = False
            if pylon_m is not None:
                brep_file = Path(brep_dir_path, f"pylon{k}_m.brep")
                export_shapes([pylon_m], str(brep_file))
                need_symmetric_engine = True
```

aidanjungo commented on ceasiompy/CPACS2GMSH/func/exportbrep.py:
```diff
@@ -0,0 +1,170 @@
+        # engine_coord = translation.get_gp_pnt().Coord()
+        engine_coord_path = Path(brep_dir_path, "engine_coord.txt")
+
+        engine_coord = (-0.1, 0.0, -0.41)
```
Should probably not be there ;)

aidanjungo commented on ceasiompy/CPACS2GMSH/func/generategmesh.py:
```diff
@@ -0,0 +1,641 @@
+    """
+    file_list = os.listdir(brep_dir_path)
+    # for now suppress nacelle center parts since they are not supported
+    file_list = [file for file in file_list if (("center" not in file))]
```
```suggestion
    file_list = [file for file in file_list if "center" not in file]
```

Commit:
test correction

aidanjungo commented on ceasiompy/CPACS2GMSH/func/generategmesh.py:
```diff
@@ -0,0 +1,641 @@
+    mesh_color_nacelle = (0, 100, 255)
+
+    mesh_color_farfield = (255, 200, 0)
+    mesh_color_symmetry = (200, 255, 0)
```
Could be constant saved at the beginig of the file or in a config file

aidanjungo replied:
or an attribute of the Class AircraftPart

aidanjungo commented on ceasiompy/CPACS2GMSH/func/generategmesh.py:
```diff
@@ -0,0 +1,641 @@
+    name : str
+        name of the part which correspond to its .brep file name
+
+    """
```
This class cloud have small functions that are called when there are loops over AircraftPart instances

aidanjungo commented on ceasiompy/CPACS2GMSH/func/wingclassification.py:
```diff
@@ -0,0 +1,746 @@
+        )
+        return True
+    else:
+        return False
```
You could test the opposite and return False, then un-indent the rest of the code

aidanjungo commented on ceasiompy/CPACS2GMSH/func/wingclassification.py:
```diff
@@ -0,0 +1,746 @@
+        (pair_points[0] not in pair_points_other) and (pair_points[1] not in pair_points_other)
+    ):
+        return False
+    else:
+        # determine the common point in the two pair of points
```
Not needed, you can un-indent the rest of the function

aidanjungo commented on ceasiompy/CPACS2GMSH/func/wingclassification.py:
```diff
@@ -0,0 +1,746 @@
+        # part of the aircraft may be deleted due to symmetry application
+        return False
+    # Search if this is the only wing section of the wing
+    if len(wing_sections) != 0:
```
```suggestion
    if wing_sections:
```

aidanjungo commented on ceasiompy/CPACS2GMSH/tests/test_exportbrep.py:
```diff
@@ -0,0 +1,73 @@
+            file_count += 1
+
+    assert file_count == 3  # simpletest_cpacs.xml containt only 3 parts
+
```
You could test that brep file have the correct names, if possible based on uid

aidanjungo commented on ceasiompy/CPACS2GMSH/tests/test_wingclassification.py:
```diff
@@ -0,0 +1,116 @@
+    # Test if the wing1_m is correctly classified
+    nb_wing_section = len(test_wingsection)
+    test_nb_section = 2
+    assert nb_wing_section == test_nb_section
```
```suggestion
    assert nb_wing_section == 2
```

aidanjungo commented on ceasiompy/CPACS2GMSH/tests/test_wingclassification.py:
```diff
@@ -0,0 +1,116 @@
+    wing_sec1 = test_wingsection[0]
+    truncated = wing_sec1["truncated"]
+    test_truncated = True
+    assert truncated == test_truncated
```
```suggestion
    assert wing_sec1["truncated"]
```

aidanjungo commented on ceasiompy/CPACS2GMSH/tests/test_wingclassification.py:
```diff
@@ -0,0 +1,116 @@
+
+    profile_1_sec1 = wing_sec1["profiles"][0]
+    test_profile_1_sec1 = [19, 20, 21]
+    assert profile_1_sec1 == test_profile_1_sec1
```
```suggestion
    assertwing_sec1["profiles"][0] == [19, 20, 21]
```

aidanjungo commented on ceasiompy/CPACS2GMSH/tests/test_wingclassification.py:
```diff
@@ -0,0 +1,116 @@
+
+    profile_2_sec1 = wing_sec1["profiles"][1]
+    test_profile_2_sec1 = [28, 30, 32]
+    assert profile_2_sec1 == test_profile_2_sec1
```
```suggestion
    assert wing_sec1["profiles"][1] == [28, 30, 32]
```

aidanjungo commented on ceasiompy/CPACS2GMSH/tests/test_wingclassification.py:
```diff
@@ -0,0 +1,116 @@
+
+    le_line_sec1 = wing_sec1["le_line"]
+    test_le_line_sec1 = [27]
+    assert le_line_sec1 == test_le_line_sec1
```
```suggestion
    assert wing_sec1["le_line"] == [27]
```

aidanjungo commented on ceasiompy/CPACS2GMSH/tests/test_wingclassification.py:
```diff
@@ -0,0 +1,116 @@
+
+    te_line_sec1 = wing_sec1["te_line"]
+    test_te_line_sec1 = [29, 31]
+    assert te_line_sec1 == test_te_line_sec1
```
```suggestion
    assert wing_sec1["te_line"] == [29, 31]
```

aidanjungo reviewed (commented):
In the overall it seems good, but I think a lot of functions could be breakdowns into small ones. It will be easier to understand and to test

Commit:
add test for get_volume_entities()

Commit:
move some methods to the aircraftpart class

Commit:
rename AircraftPart for ModelPart

Commit:
add test for ModelPart methods

Commit:
remove beta engine_pos determination from exportbrep.py

Commit:
replace childs by children

Commit:
add test and correct minor codestyle errors

Commit:
add test for wingclassification  and correct minor codestyle errors

Commit:
minor codestyle errors

Commit:
minor codestyle errors

Commit:
minor codestyle errors

Commit:
minor codestyle errors again

Commit:
simplify tests

Commit:
correct codestyle

Commit:
correct codestyle

Commit:
pyflake correction

Commit:
add test for classify_wing_section()

Commit:
add test for classify_special_section()

Commit:
simplify exportbrep.py

Commit:
small correction

Commit:
edit readme of CPACS2GMSH module

Commit:
edit readme of CPACS2GMSH module

Commit:
Update README.md

Commit:
Update README.md

typos

Commit:
small updates

Commit:
small updates and corrections

Commit:
update Farfield marker + small corrections

Commit:
update Farfield marker in test

Commit:
small change on meshbox coef

aidanjungo reviewed (commented):
ok
